### PR TITLE
285, ImageMosaic excessive memory usage when mosaicking many little images

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: jt-mosaic/src/main/java/it/geosolutions/jaiext/mosaic/MosaicOpImage.java [hunk 1]
```diff
@@ -817,14 +822,21 @@ public Raster computeTile(int tileX, int tileY) {
         ColorModel[] alphaChannelColorModels = new ColorModel[numSources];
         // The previous array is filled with the source raster data
         int intersectingSourceCount = 0;
+
+        // remove this!
```
Comment (dromagnoli) [inline jt-mosaic/src/main/java/it/geosolutions/jaiext/mosaic/MosaicOpImage.java:826]: Does it need to be removed? :)

File: jt-mosaic/src/main/java/it/geosolutions/jaiext/mosaic/PixelIterator.java [hunk 2]
```diff
@@ -0,0 +1,303 @@
+/* JAI-Ext - OpenSource Java Advanced Image Extensions Library
+ *    http://www.geo-solutions.it/
+ *    Copyright 2014 - 2015 GeoSolutions
```
Comment (dromagnoli) [inline jt-mosaic/src/main/java/it/geosolutions/jaiext/mosaic/PixelIterator.java:3]:
```suggestion
 *    Copyright 2022 GeoSolutions
```

File: jt-mosaic/src/test/java/it/geosolutions/jaiext/mosaic/PixelIteratorTest.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
+/* JAI-Ext - OpenSource Java Advanced Image Extensions Library
+ *    http://www.geo-solutions.it/
+ *    Copyright 2014 GeoSolutions
```
Comment (dromagnoli) [inline jt-mosaic/src/test/java/it/geosolutions/jaiext/mosaic/PixelIteratorTest.java:3]:
```suggestion
 *    Copyright 2022 GeoSolutions
```

**Review by dromagnoli — APPROVED** [review]
Looks good to me.
I was about to suggest for a rename of a method but then I have noticed that the referred  variable was part of the old code so no need...

(I was talking about alphaPresentinRaster and related getter.
If we want to make a Camel, happy :), we could rename it to alphaPresentInRaster but that's not a big deal)

File: jt-mosaic/src/main/java/it/geosolutions/jaiext/mosaic/MosaicOpImage.java (hunk 1 above)
Comment (aaime) [reply jt-mosaic/src/main/java/it/geosolutions/jaiext/mosaic/MosaicOpImage.java:826]: Yup!

**Comment by aaime** [comment]
Replaced alphainraster with a simpler "hasAlpha", fixed the other bits.

**Pull request merged**